curriculum/challenges/english/25-front-end-development/review-react-forms-data-fetching-and-routing/67c9e932c6c4234532d46394.md

```diff
@@ -0,0 +1,424 @@
+
+- **Multiple Views and Route Setup**: It is common in larger applications to have multiple views and routes setup like this:
+
+```jsx
+<Routes>
+  <Route index element={<Home />} />
+  <Route path="about" element={<About />} />
+
+  <Route path="products">
+    <Route index element={<ProductsHome />} />
+    <Route path=":category" element={<Category />} />
+    <Route path=":category/:productId" element={<ProductDetail />} />
+    <Route path="trending" element={<Trending />} />
+  </Route>
+</Routes>
+```
+
+The `index` prop in these examples is meant to represent the default route for a given path segment. So the `Home` component will be shown at the root `/` path while the `ProductsHome` component will be shown at the `/products` path.
+
+- **Nesting Routes**: You can nest routes inside other routes which results in the path of the child route being appended to the parent route's path. 
+
+```jsx
+<Route path="products">
+  <Route path="trending" element={<Trending />} />
+</Route>
+```
+
+In the example above, the path for the trending products will be products/trending.
+
+- **Dynamic Segments**: A dynamic segment is where any part of the URL path is dynamic. 
+
+```jsx
+<Route path=":category" element={<Category />} />
+```
+
+In this example we have a dynamic segment called `category`. When a user navigates to a URL like products/brass-instruments, then the view will change to the `Category` component and you can dynamically fetch the appropriate data based on the segment.
+
+- **`useParams` Hook**: This hook is used to access the dynamic parameters from a URL path.
+
+```js
+import { useParams } from "react-router";
+
+export default function Category() {
+  let params = useParams();
+   {/* Accessing the category param: params.category */}
+   {/* rest of code goes here */}
+}
+```
+
+## React Frameworks
+
+- **Introduction**: React frameworks provide features like routing, image optimizations, data fetching, authentication and more. This means that you might not need to set up separate frontend and backend applications for certain use cases. Examples of React Frameworks include Next.js and Remix.
+- **Next.js Routing**: This routing system includes support for dynamic routes, parallel routes, route handlers, redirects, internalization and more.
+
+Here is an example of creating a custom request handler:
+
+```js
+export async function GET() {
+  const res = await fetch("https://example-api.com");
+  const data = await res.json();
+
+  return Response.json({ data });
+}
+```
+
+- **Next.js Image Optimization**: The `Image` component extends the native HTML `img` element and allows for faster page loads and size optimizations. This means that images will only load when they enter the viewport and the `Image` component will automatically serve correctly sized images for each device.
```

Comment:
```suggestion
- **Next.js Image Optimization**: The `Image` component extends the native HTML `img` element and allows for faster page loading and size optimizations. This means that images will only load when they enter the viewport and the `Image` component will automatically serve correctly sized images for each device.
```

Reply:
"Page loads" is valid and grammatically correct.